src/wui/interactive_gamebase.cc

```diff
@@ -282,12 +282,24 @@ void InteractiveGameBase::rebuild_gamespeed_menu() {
 void InteractiveGameBase::gamespeed_menu_selected(GameSpeedEntry entry) {
 	switch (entry) {
 	case GameSpeedEntry::kIncrease: {
-		increase_gamespeed();
+		if (SDL_GetModState() & KMOD_SHIFT) {
```

Comment:
Personally, I would find this more readable:
`increase_gamespeed(SDL_GetModState() & KMOD_SHIFT? 250 : SDL_GetModState() & KMOD_CTRL ? 10000 : 250);`
Also please make the three values `constexpr`.